torchinductor/sizevars.py

```diff
@@ -314,7 +320,8 @@ def strideof(name):
             code.writeline(f"{name}_stride = {name}.stride()")
             return f"{name}_stride"
 
-        needed = set(map(str, self.var_to_val.keys())) - set(self.replacements.keys())
+        needed_vars = set(self.var_to_val.keys()) - set(self.replacements.keys())
```

Comment:
@jansel I think this was previously just a bug? `self.replacements` is a dict from `sympy.Symbol` to `sympy.Expr`. And here you're deduplicating against a set with strings.

Reply:
I think you are correct

Reply:
Seems like fixing this bug causes some issues in `test/test_torchinductor.py::GpuTests::test_triton_conv_cuda`.

I get a `NameError: name 's0' is not defined` (cc: @pyjhzwh )

Reply:
Gonna revert to the old behavior for now and add a note.